packages/@aws-cdk/aws-s3/lib/notifications-resource/lambda-source/src/index.py

```diff
@@ -0,0 +1,94 @@
+import boto3  # type: ignore
+import json
+import urllib.request
+from typing import List
+
+s3 = boto3.client("s3")
+cf = boto3.client("cloudformation")
+
+
+def handler(event: dict, context):
+    response_status = "SUCCESS"
+    error_message = ""
+    try:
+        props = event["ResourceProperties"]
+        bucket = props["BucketName"]
+        if is_rollback_in_progress(event):
+            return
+        s3.put_bucket_notification_configuration(
+            Bucket=bucket,
+            NotificationConfiguration=get_config(
+                current_config=s3.get_bucket_notification_configuration(Bucket=bucket),
+                new_config=props["NotificationConfiguration"],
+                old_config=event.get("OldResourceProperties", {}).get("NotificationConfiguration", {}),
+                request_type=event["RequestType"],
+            ),
+        )
+    except Exception as e:
+        print("Failed to put bucket notification configuration:", e)
+        response_status = "FAILED"
+        error_message = f"Error: {str(e)}. "
+    finally:
+        submit_response(event, context, response_status, error_message)
+
+
+def is_rollback_in_progress(event: dict) -> bool:
```

Comment:
Maybe this should be called `skip_put_bucket_notification_configuration`? or something along those lines.